rosidl_typesupport_connext_c/resource/msg__type_support_c.cpp.em

```diff
@@ -453,24 +453,31 @@ _@(message.structure.namespaced_type.name)__to_cdr_stream(
     fprintf(stderr, "failed to call @(__dds_cpp_msg_type_prefix)_Plugin_serialize_to_cdr_buffer()\n");
     return false;
   }
-  cdr_stream->buffer_length = expected_length;
-  if (cdr_stream->buffer_length > (std::numeric_limits<unsigned int>::max)()) {
-    fprintf(stderr, "cdr_stream->buffer_length, unexpectedly larger than max unsigned int\n");
+  if (expected_length > (std::numeric_limits<unsigned int>::max)()) {
+    fprintf(stderr, "expected_length, unexpectedly larger than max unsigned int\n");
     return false;
   }
-  if (cdr_stream->buffer_capacity < cdr_stream->buffer_length) {
+  if (cdr_stream->buffer_capacity < expected_length) {
+    uint8_t * new_buffer = static_cast<uint8_t *>(cdr_stream->allocator.allocate(expected_length, cdr_stream->allocator.state));
+    if (NULL == new_buffer) {
+      fprintf(stderr, "failed to allocate memory for cdr data\n");
+      return false;
+    }
     cdr_stream->allocator.deallocate(cdr_stream->buffer, cdr_stream->allocator.state);
-    cdr_stream->buffer = static_cast<uint8_t *>(cdr_stream->allocator.allocate(cdr_stream->buffer_length, cdr_stream->allocator.state));
+    cdr_stream->buffer = new_buffer;
+    cdr_stream->buffer_capacity = expected_length;
```

Comment:
why not using `rcutils_uint8_array_resize` here?
https://github.com/ros2/rcutils/blob/master/src/uint8_array.c#L74

Reply:
I didn't plan to refactor existing code - just fix the failing test / broken logic.

I would suggest refactoring it in a follow up PR.

Reply:
I don't think it has anything to do with refactoring. The introduced code here is really the same as the one in rcutils: https://github.com/ros2/rcutils/blob/master/src/uint8_array.c#L91-L98 which also takes care of resetting the length and capacity in case the allocation fails.

Reply:
> The introduced code here is really the same as the one in rcutils:

My patch doesn't introduce much at all. It essentially just adds the missing `cdr_stream->buffer_capacity = expected_length;` assignment.

Also using `rcutils_uint8_array_resize` implies that the original buffer is altered in case the reallocation fails which is neither necessary nor (imo) desired.